charts/argo-cd/values.yaml

```diff
@@ -46,6 +46,9 @@ global:
   additionalLabels: {}
     # app: argo-cd
 
+  # -- Number of old deployment ReplicaSets to retain. The rest will be garbage collected.
+  revisionHistoryLimit: 3
```

Comment:
Lets set it to 5 again so we can also update the STS of the controller.

Reply:
STS does not support this - it's there from historic reasons because it was Deployment 1 year ago.

Reply:
Also if you try to remove history limit from STS you get following:

```sh
The StatefulSet "argocd-application-controller" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', 'updateStrategy', 'persistentVolumeClaimRetentionPolicy' and 'minReadySeconds' are forbidden
```

Only way to upgrade is to run these and I wanted to keep this for 6.x release when we remove old things.

```sh
kubectl delete argocd-application-controller
helm upgrade argocd
```

Reply:
Ah you are right 👍

Reply:
Any reason why we're not matching 5 here as a default?

Reply:
I've reduced this to 3 to make UI less cluttered when Argo CD is watching itself. Users can adjust amount of replicas so if you want more you can adjust it.

Reply:
I'm happy with this, but I'd try to align to previous or upstream defaults if we can.